tests/infra/types.py

```diff
@@ -0,0 +1,39 @@
+# SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Callable, Mapping, Optional, Sequence, Union
+
+import jax
+from flax import linen, nnx
+
+# Convenience alias. Could be used to represent jax.Array, torch.Tensor, np.ndarray, etc.
+Tensor = Union[jax.Array]
+
+# Convenience alias. Could be used to represent nnx.Module, torch.nn.Module, etc.
+# NOTE nnx.Module is the newest API, linen.Module is legacy but it is used in some
+# huggingface models.
+Model = Union[nnx.Module, linen.Module]
+
+
+@dataclass
+class Workload:
```

Comment:
Having this class defined in `types.py` somehow doesn't seem right, maybe move to a separate module?